g5k_discovery/vue/capabilities.js

```diff
@@ -84,14 +88,47 @@ export const advancedCapabilities = {
       },
       NVMe: {
         capability({ storageDevices }) {
-          return storageDevices.some(({ driver, interface: iface }) =>
-            driver === "nvme" || (iface || "").toLowerCase() === "pcie"
+          return storageDevices.some(
+            ({ driver, interface: iface }) =>
+              driver === "nvme" || (iface || "").toLowerCase() === "pcie"
           )
             ? "Yes"
             : "No";
         },
         tagPrefix: "NVMe: ",
       },
     },
-  }
+  },
+  RDMA: {
+    custom: {
+      InfiniBand: {
+        capability({ infiniband }) {
+          return infiniband ? "Yes" : "No";
+        },
+      },
+      GPUDirect: {
+        capability({ infiniband, gpu }) {
+          let ib_present = infiniband;
+          let gpu_present = false;
+          if (gpu) {
+            if (gpu.gpuModel) {
+              gpu_present = gpu.gpuModel.toLowerCase() in GPUDIRECT_MODELS;
+              console.log(gpu);
+            }
+          }
+          return ib_present && gpu_present ? "Yes" : "No";
```

Comment:
TIL checking for undefined variables like this is totally fine in JavaScript 🤯

Reply:
undefined is "falsy" apparently. Please don't take any of this as gospel though, I poked it until it worked

Reply:
Even better ;), this function could be rewritten as:

```typescript
capability({ infiniband, gpu }) {
  return infiniband && gpu && gpu.gpuModel && gpu.gpuModel.toLowerCase() in GPUDIRECT_MODELS ? "Yes" : "No"
}
```

This is because `&&` is a guard operator and it evaluates lazily; it won't execute anything in the right branch if the left branch evaluates to falsy.

Reply:
> undefined is "falsy" apparently. Please don't take any of this as gospel though, I poked it until it worked

I did Google it to make sure before commenting, and it does seem this is a fairly common pattern in JS. I think Jason's change makes more sense in the general case. It also makes the "other languages" part of my brain make a happy face.